From a70ef13ff0e9d22e1c93a1563a24d59e20ff726e Mon Sep 17 00:00:00 2001
From: Luoyouming <luoyouming@huawei.com>
Date: Thu, 27 Oct 2022 18:36:56 +0800
Subject: [PATCH 0997/1646] [Backport] Revert "RDMA/hns: Support cqe inline in
 user space"

driver inclusion
category: cleanup
bugzilla: https://gitee.com/openeuler/kernel/issues/I5USIG

Reference: https://gitee.com/openeuler/kernel/commit/36114286e355c7471eb87e866706c1cb157ea02f

--------------------------------

This reverts commit 5a18385d182db24e95c8f4ce9fb0e34ed6139803.

Fixes: 5a18385d182d ("RDMA/hns: Support cqe inline in user space")

Signed-off-by: Luoyouming <luoyouming@huawei.com>
Reviewed-by: Yangyang Li <liyangyang20@huawei.com>
Reviewed-by: Haoyue Xu <xuhaoyue1@hisilicon.com>
Reviewed-by: YueHaibing <yuehaibing@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
---
 drivers/infiniband/hw/hns/hns_roce_device.h |  1 -
 drivers/infiniband/hw/hns/hns_roce_hw_v2.c  | 12 ------------
 drivers/infiniband/hw/hns/hns_roce_hw_v2.h  |  3 +--
 drivers/infiniband/hw/hns/hns_roce_main.c   |  4 ----
 include/uapi/rdma/hns-abi.h                 |  1 -
 5 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/infiniband/hw/hns/hns_roce_device.h b/drivers/infiniband/hw/hns/hns_roce_device.h
index 218d66166b05..ab052b54e206 100644
--- a/drivers/infiniband/hw/hns/hns_roce_device.h
+++ b/drivers/infiniband/hw/hns/hns_roce_device.h
@@ -145,7 +145,6 @@ enum {
 	HNS_ROCE_CAP_FLAG_DIRECT_WQE		= BIT(12),
 	HNS_ROCE_CAP_FLAG_SDI_MODE		= BIT(14),
 	HNS_ROCE_CAP_FLAG_STASH			= BIT(17),
-	HNS_ROCE_CAP_FLAG_CQE_INLINE		= BIT(19),
 	HNS_ROCE_CAP_FLAG_RQ_INLINE		= BIT(20),
 };
 
diff --git a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
index de87319ec202..12c7a618794d 100644
--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
@@ -4727,18 +4727,6 @@ static int modify_qp_init_to_rtr(struct ib_qp *ibqp,
 		hr_reg_clear(qpc_mask, QPC_RQIE);
 	}
 
-	if (udata &&
-	    (ibqp->qp_type == IB_QPT_RC || ibqp->qp_type == IB_QPT_XRC_TGT) &&
-	    (uctx->config & HNS_ROCE_ALLOC_UCTX_CQE_INLINE_FLAGS)) {
-		hr_reg_write_bool(context, QPC_CQEIE,
-				  hr_dev->caps.flags &
-				  HNS_ROCE_CAP_FLAG_CQE_INLINE);
-		hr_reg_clear(qpc_mask, QPC_CQEIE);
-
-		hr_reg_write(context, QPC_CQEIS, 0);
-		hr_reg_clear(qpc_mask, QPC_CQEIS);
-	}
-
 	return 0;
 }
 
diff --git a/drivers/infiniband/hw/hns/hns_roce_hw_v2.h b/drivers/infiniband/hw/hns/hns_roce_hw_v2.h
index dbb19fe680c9..16437df320ea 100644
--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.h
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.h
@@ -529,8 +529,7 @@ struct hns_roce_v2_qp_context {
 #define QPC_RQ_RTY_TX_ERR QPC_FIELD_LOC(607, 607)
 #define QPC_RX_CQN QPC_FIELD_LOC(631, 608)
 #define QPC_XRC_QP_TYPE QPC_FIELD_LOC(632, 632)
-#define QPC_CQEIE QPC_FIELD_LOC(633, 633)
-#define QPC_CQEIS QPC_FIELD_LOC(634, 634)
+#define QPC_RSV3 QPC_FIELD_LOC(634, 633)
 #define QPC_MIN_RNR_TIME QPC_FIELD_LOC(639, 635)
 #define QPC_RQ_PRODUCER_IDX QPC_FIELD_LOC(655, 640)
 #define QPC_RQ_CONSUMER_IDX QPC_FIELD_LOC(671, 656)
diff --git a/drivers/infiniband/hw/hns/hns_roce_main.c b/drivers/infiniband/hw/hns/hns_roce_main.c
index 2d59e0bebf98..0b04eb196fa2 100644
--- a/drivers/infiniband/hw/hns/hns_roce_main.c
+++ b/drivers/infiniband/hw/hns/hns_roce_main.c
@@ -383,10 +383,6 @@ static int hns_roce_alloc_ucontext(struct ib_ucontext *uctx,
 		context->config |= ucmd.config & HNS_ROCE_ALLOC_UCTX_RQ_INLINE_FLAGS;
 		if (hr_dev->caps.flags & HNS_ROCE_CAP_FLAG_RQ_INLINE)
 			resp.config |= HNS_ROCE_ALLOC_UCTX_RQ_INLINE_FLAGS;
-
-		context->config |= ucmd.config & HNS_ROCE_ALLOC_UCTX_CQE_INLINE_FLAGS;
-		if (hr_dev->caps.flags & HNS_ROCE_CAP_FLAG_CQE_INLINE)
-			resp.config |= HNS_ROCE_ALLOC_UCTX_CQE_INLINE_FLAGS;
 	}
 
 	ret = hns_roce_uar_alloc(hr_dev, &context->uar);
diff --git a/include/uapi/rdma/hns-abi.h b/include/uapi/rdma/hns-abi.h
index 7d09da8d99ce..90d1c3629b7b 100644
--- a/include/uapi/rdma/hns-abi.h
+++ b/include/uapi/rdma/hns-abi.h
@@ -88,7 +88,6 @@ struct hns_roce_ib_create_qp_resp {
 enum {
 	HNS_ROCE_UCONTEXT_EXSGE_CALC_MODE = 1 << 0,
 	HNS_ROCE_ALLOC_UCTX_RQ_INLINE_FLAGS = 1 << 1,
-	HNS_ROCE_ALLOC_UCTX_CQE_INLINE_FLAGS = 1 << 2,
 };
 
 struct hns_roce_ib_alloc_ucontext_resp {
-- 
2.17.1

